### ISSUE TITLE
error with class parameter in plotDataCI and plotDataBoxplot

### ISSUE THREAD

**Issue by Quentin62** [modal-inria/MixtComp#1]
- with `class = 2` the class number 2 has not the same color as the class number 2 when `class = NULL` in plotDataCI and plotDataBoxplot
- with `class = 2` the class number 2 has a wrong position in plotDataCI
- using both `grl = TRUE` and `class = 2` generates an error in plotDataCI

``` r
library(RMixtComp)

data(simData)

algo <- list(
  nInd = 100,
  nbBurnInIter = 100,
  nbIter = 100,
  nbGibbsBurnInIter = 100,
  nbGibbsIter = 100,
  nInitPerClass = 3,
  nSemTry = 20,
  confidenceLevel = 0.95,
  ratioStableCriterion = 0.95,
  nStableCriterion = 10
  )

resLearn <- mixtCompLearn(simData$dataLearn$data.frame[,3:4], simData$model$supervised[3:4], nClass = 2, nRun = 3)

plotDataBoxplot(resLearn, "Gaussian1", grl = FALSE)
plotDataBoxplot(resLearn, "Gaussian1", class = 1, grl = FALSE) 
plotDataBoxplot(resLearn, "Gaussian1", class = 2, grl = FALSE) # wrong color

plotDataBoxplot(resLearn, "Gaussian1", grl = TRUE)
plotDataBoxplot(resLearn, "Gaussian1", class = 1, grl = TRUE) # wrong color 
plotDataBoxplot(resLearn, "Gaussian1", class = 2, grl = TRUE) # wrong color


plotDataCI(resLearn, "Gaussian1", grl = FALSE)
plotDataCI(resLearn, "Gaussian1", class = 1, grl = FALSE) #  wrong position
plotDataCI(resLearn, "Gaussian1", class = 2, grl = FALSE) # wrong color + wrong position

plotDataCI(resLearn, "Gaussian1", grl = TRUE)
plotDataCI(resLearn, "Gaussian1", class = 1, grl = TRUE) # error
plotDataCI(resLearn, "Gaussian1", class = 2, grl = TRUE) # error




plotDataBoxplot(resLearn, "Categorical1", grl = FALSE)
plotDataBoxplot(resLearn, "Categorical1", class = 1, grl = FALSE) 
plotDataBoxplot(resLearn, "Categorical1", class = 2, grl = FALSE) # wrong color

plotDataBoxplot(resLearn, "Categorical1", grl = TRUE)
plotDataBoxplot(resLearn, "Categorical1", class = 1, grl = TRUE) # wrong color 
plotDataBoxplot(resLearn, "Categorical1", class = 2, grl = TRUE) # wrong color


plotDataCI(resLearn, "Categorical1", grl = FALSE)
plotDataCI(resLearn, "Categorical1", class = 1, grl = FALSE) #  wrong position
plotDataCI(resLearn, "Categorical1", class = 2, grl = FALSE) # wrong color + wrong position

plotDataCI(resLearn, "Categorical1", grl = TRUE)
plotDataCI(resLearn, "Categorical1", class = 1, grl = TRUE) # error
plotDataCI(resLearn, "Categorical1", class = 2, grl = TRUE) # error
```

**Comment by Quentin62** [comment]
- [ ] PlotDataCI + numerical
- [ ] PlotDataCI + categorical
- [ ] PlotDataBoxplot + numerical
- [ ] PlotDataBoxplot + categorical

**Comment by Quentin62** [comment]
correction will be released in RMixtCompUtilities 4.1.3
